contrib/test/integration/main.yml

```diff
@@ -0,0 +1,118 @@
+- hosts: all
+  remote_user: root
+  vars:
+    xunit: false
+  tasks:
+    - name: set up the system
+      include: system.yml
+
+    - name: install Golang tools
+      include: golang.yml
+
+    - name: clone build and install bats
+      include: "build/bats.yml"
+      async: 2400
+      poll: 0
+      register: bats_task
+
+    - name: clone build and install cri-tools
+      include: "build/cri-tools.yml"
+      async: 2400
+      poll: 0
+      register: cri_tools_task
+
+    - name: clone build and install cri-o
+      include: "build/cri-o.yml"
+      async: 2400
+      poll: 0
+      register: crio_task
+
+    - name: clone build and install kubernetes
+      include: "build/kubernetes.yml"
+      async: 2400
+      poll: 0
+      register: kubernetes_task
+
+    - name: clone build and install runc
+      include: "build/runc.yml"
+      async: 2400
+      poll: 0
+      register: runc_task
+
+    - name: clone build and install networking plugins
+      include: "build/plugins.yml"
+      async: 2400
+      poll: 0
+      register: plugins_task
+
+    - name: wait for bats build to be done
```

Comment:
@sdodson I'm seeing this error:

```
TASK [wait for bats build to be done] ***************************************************************************************************
task path: /data/src/github.com/kubernetes-incubator/cri-o/contrib/test/integration/main.yml:48
fatal: [localhost]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'bats_task' is undefined\n\nThe error appears to have been in '/data/src/github.com/kubernetes-incubator/cri-o/contrib/test/integration/main.yml': line 48, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: wait for bats build to be done\n      ^ here\n"}
```

Any clue why? I don't seem to be able to trigger it with this:

```yaml
- hosts: all
  become_user: root
  tasks:
    - name: clone build and install bats
      command: "sleep 10"
      async: 2400
      poll: 0
      register: bats_task

    - name: clone build and install cri-tools
      command: "sleep 20"
      async: 2400
      poll: 0
      register: cri_tools_task

    - name: wait for bats build to be done
      async_status:
        jid: "{{ bats_task.ansible_job_id }}"
      register: async_poll_results
      until: async_poll_results.finished
      retries: 240
      delay: 10

    - name: wait for cri-tools build to be done
      async_status:
        jid: "{{ cri_tools_task.ansible_job_id }}"
      register: async_poll_results
      until: async_poll_results.finished
      retries: 240
      delay: 10
```

I don't get how that fact can be unset... it's an unconditional `register`

Reply:
I bet include: doesn't register anything. @mtnbikenc what do you think?

Reply:
register on whatever task actually takes a while assuming subsequent tasks between that and the wait can execute async